src/iuse.cpp

```diff
@@ -8178,6 +8224,90 @@ static bool multicooker_hallu( player &p )
 
 }
 
+int iuse::autoclave( player *p, item *it, bool t, const tripoint &pos )
+{
+    if( t ) {
+        if( !it->units_sufficient( *p ) ) {
+            add_msg( m_bad, _( "The autoclave ran out of battery and stopped before completing its cycle." ) );
+            it->active = false;
+            item *clean_cbm = nullptr;
+            for( item &bio : it->contents ) {
+                if( bio.is_bionic() ) {
+                    clean_cbm = &bio;
+                }
+            }
+            if( clean_cbm ) {
+                g->m.add_item( pos, *clean_cbm );
+                it->remove_item( *clean_cbm );
+            }
+            return 0;
+        }
+
+        int Cycle_time = it->get_var( "CYCLETIME", 0 );
+        Cycle_time -= 1;
+        if( Cycle_time <= 0 ) {
+            it->active = false;
+            it->erase_var( "CYCLETIME" );
+            item *clean_cbm = nullptr;
+            for( item &bio : it->contents ) {
+                if( bio.is_bionic() ) {
+                    bio.unset_flag( "NO_STERILE" );
+                    bio.set_var( "sterile", 1 ); // sterile for 1s if not (packed);
+                    clean_cbm = &bio;
+                }
+            }
+            if( clean_cbm ) {
+                g->m.add_item( pos, *clean_cbm );
+                it->remove_item( *clean_cbm );
+            }
+
+        } else {
+            it->set_var( "CYCLETIME", Cycle_time );
+        }
+
+    } else if( !it->active ) {
+        if( p->is_underwater() ) {
+            p->add_msg_if_player( m_info, _( "You can't do that while underwater." ) );
+            return 0;
+        }
+
+        auto reqs = *requirement_id( "autoclave_item" );
+
+        const item_location to_sterile = game_menus::inv::sterilize_cbm( *p );
+
+        if( !to_sterile ) {
+            return 0;
+        }
+
+        if( query_yn( _( "Start the autoclave?" ) ) ) {
+
+            for( const auto &e : reqs.get_components() ) {
+                p->consume_items( e, 1, is_crafting_component );
+            }
+            for( const auto &e : reqs.get_tools() ) {
+                p->consume_tools( e );
+            }
+            p->invalidate_crafting_inventory();
+            const item *cbm = to_sterile.get_item();
+
+            std::vector<item_comp> comps;
+            comps.push_back( item_comp( cbm->typeId(), 1 ) );
+            p->consume_items( comps, 1, is_crafting_component );
```

Comment:
This may actually consume a completely different item that has the same id (because it only checks the id, because that's the only thing it knows about).

Reply:
How to remove the CBM then? This is how install cbm does it so I might need to change it there also.

Reply:
Well,  `item_location` (the type of `to_sterile` offers the `remove_item` function.

What function / line / source file is "This is how install cbm does it"?

Reply:
Iexamine line 4289 and npctalk_func line 373 are isong consume_items to remove the CBM you chose to install.